scribejava-core/src/main/java/com/github/scribejava/core/oauth/OAuth20Service.java

```diff
@@ -34,7 +38,18 @@ public OAuth20Service(DefaultApi20 api, OAuthConfig config) {
 
     //sync version, protected to facilitate mocking
     protected OAuth2AccessToken sendAccessTokenRequestSync(OAuthRequest request) throws IOException {
-        return api.getAccessTokenExtractor().extract(request.send().getBody());
+        final Response response = request.send();
+        final TokenExtractor<OAuth2AccessToken> extractor = api.getAccessTokenExtractor();
+        if (response.getCode() != 200) {
+            if (extractor instanceof OAuth2AccessTokenJsonExtractor) {
+                final OAuth2AccessTokenJsonExtractor jsonExtractor = (OAuth2AccessTokenJsonExtractor) extractor;
+                throw new OAuthRequestException("Get Access token error",
+                        jsonExtractor.extractError(response.getBody()));
+            } else {
+                throw new OAuthException("Get Access token error: " + response.getBody());
```

Comment:
I think the important part here is the http return code.

In the `extract` method you don't have this information. So, the only solution is trying to parse the response as a valid one, then trying to parse it as an error one. 
I'm not feeling it good.